src/app/_components/ProfilePageHeader/ProfilePageHeader.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import {
+  Avatar,
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Typography,
+  useTheme,
+} from "@mui/material";
+
+interface ProfilePageHeaderProps {
+  name: string;
+  username: string;
+  bio: string;
+  avatarUrl: string; // for demonstration purpose only
+  onEditProfile: () => void;
+}
+
+const ProfilePageHeader: React.FC<ProfilePageHeaderProps> = ({
+  name,
+  username,
+  bio,
+  avatarUrl,
+  onEditProfile,
+}) => {
+  const theme = useTheme();
+
+  const contentText = (text: string | undefined | null, maxLength: number) => {
+    if (!text) return "";
+    return text.length > maxLength
+      ? `${text.substring(0, maxLength)}...`
+      : text;
+  };
```

Comment:
Cool function! Should prob go into a utility class instead of being here for reusability.
And if you use `noWrap` on `<Typography>` and make sure it's parent has a width limit then it will automatically ellipsis.